server/common/config/client_config.py

```diff
@@ -29,21 +42,29 @@ def get_client_config(app_config: AppConfig, data_adaptor: Dataset, current_app)
         "about-dataset": about,
     }
 
+    # Check if authentication is disabled
+    auth_disabled = os.environ.get("CELLXGENE_DISABLE_AUTH", "false").lower() in ("true", "1", "yes")
+
     # parameters
     parameters = {
```

Comment:
Server config note: these new flags drive everything on the client. When auth is disabled the UI hides autosave and keeps annotations/genesets read-only; otherwise it exposes the menus and autosave timer. The `annotations-user-data-idhash` is used for naming autosave files on the backend.